src/Core/AttributeGroup/AttributeGroupViewDataProvider.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\AttributeGroup;
+
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Exception\AttributeGroupConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Query\CheckIsColorGroupById;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Query\GetAttributeGroupNameById;
+
+/**
+ * Provides data required for attribute group view action
+ */
+final class AttributeGroupViewDataProvider
+{
+    /**
+     * @var CommandBusInterface
+     */
+    private $queryBus;
+
+    /**
+     * @var int
+     */
+    private $contextLangId;
+
+    /**
+     * @param CommandBusInterface $queryBus
+     * @param int $contextLangId
+     */
+    public function __construct(CommandBusInterface $queryBus, $contextLangId)
+    {
+        $this->queryBus = $queryBus;
+        $this->contextLangId = $contextLangId;
+    }
+
+    /**
+     * @param int $attributeGroupId
+     *
+     * @throws AttributeGroupConstraintException
+     */
+    public function isColorGroup($attributeGroupId)
+    {
+        return $this->queryBus->handle(new CheckIsColorGroupById($attributeGroupId));
```

Comment:
Why this adapter is needed? Can't you use queries without adapter? Otherwise, they don't add much value imo. @matks wdyt?

Reply:
So far we have been using adapter DataProviders which do not rely on CQRS for grid

Although the reason behind are not clear (the main reason being "we started like it and we dont know yet whether CQRS is valid for this usecase*") let's keep doing it 😉so please move this `AttributeGroupViewDataProvider` into Adapter namespace and use legacy ObjectModel to obtain the required data

*Might be solved by @sarjon suggestion about read data models